commands/addons.js

```diff
@@ -255,36 +254,45 @@ module.exports = {
 
     run:         cli.command(co.wrap(run)),
     usage:       `${topic} [--all|--app APP]`,
-    description: 'lists your add-ons and attachments',
+    description: 'Lists your add-ons and attachments',
     help:        `The default filter applied depends on whether you are in a Heroku app
 directory. If so, the --app flag is implied. If not, the default of --all
 is implied. Explicitly providing either flag overrides the default
-behaviour.
+behavior.
 
 Examples:
 
   $ heroku ${topic} --all
-  $ heroku ${topic} --app acme-inc-website
-
-Identifying specific add-ons:
-
-  Add-ons have canonical names (ADDON_NAME in these command help texts).
-  They also application-specific names called aliases or attachments. In
-  many cases, these names can be used interchangeably when unambiguous.
-
-  For example, given a fictional \`slowdb\` add-on named \`slowdb-cubed-1531\`
-  attached as \`FOO_DB\` to \`app1\` and BAR_DB to \`app2\`, the following
-  invocations are considered equivalent:
-
-  $ heroku addons:upgrade slowdb             slowdb:premium
-  $ heroku addons:upgrade --app app1 FOO_DB  slowdb:premium
-  $ heroku addons:upgrade app1::FOO_DB       slowdb:premium
-  $ heroku addons:upgrade app2::BAR_DB       slowdb:premium
-  $ heroku addons:upgrade acme-inc-datastore slowdb:premium
-
-  If the name used is ambiguous (e.g. if you used \`slowdb\` with more than
-  one add-on of that type installed), the command will error due to
-  ambiguity and a more specific identifier will need to be chosen.
+  $ heroku ${topic} --app acme-inc-www
+
+Overview of Add-ons:
+
+  Add-ons are created with the \`addons:create\` command, providing a reference
+  to an add-on service (such as \`heroku-postgresql\`) or a service and plan
+  (such as \`heroku-postgresql:hobby-dev\`).
+
+  At creation, each add-on is assigned a globally unique name (or one can be
+  provided by the user). Each add-on has at least one attachment alias to each
+  application which uses the add-on. In all cases, the owning application will
+  be attached to the add-on. An attachment alias is unique to its application,
+  and is used as a prefix to any environment variables it exports to the
+  application.
+
+  For instance, a \`heroku-postgresql:hobby-dev\` add-on named \`www-db\`, which
+  is owned by \`acme-inc-www\` and shared with \`acme-inc-dwh\` is represented
+  below:
+
+    $ heroku addons --app acme-inc-www
+    Add-on                               Plan       Price
+    ───────────────────────────────────  ─────────  ──────
+    heroku-postgresql (www-db)           hobby-dev  free
+     ├─ as DATABASE
+     └─ as WWW_DB on acme-inc-dwh app
+
+  In that example, \`DATABASE\` is an attachment alias on the owning application
+  and exports \`DATABASE_URL\` to that application while \`WWW_DB\` is an
+  attachment alias to the \`acme-inc-dwh\` application and shares the add-on to
+  that application by exporting \`WWW_DB_URL\` to it.
```

Comment:
Going with "environment variable" since that's what I've used elsewhere in the doc and because I believe we should make an effort to separate "config vars" (things users set on an app) and "add-on vars", both of which are exported *as* environment vars. If you think we should keep it as "config var" I could be persuaded though :)